### PR TITLE
fix(search): fix profile popup opened on search title clicked

### LINKED ISSUE #4795: the search is showing chats as search results and clicking on it shows the view profile modal
# Bug Report

## Description

the search is showing chats as search results (do the designs have this?) and when clicking on the chat it shows a view profile modal

expected: it shouldn't show the view profile modal for sure since it's a chat not a profile, but unclear what is the behaviour here, I don't recall seeing this in the designs

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #4795

Makes sure to only open the profile popup if the click is a pubkey.
Also opens the popup on ctrl+f

### CODE REVIEW

File: ui/imports/utils/Utils.qml
Comment (jrainville) [reply ui/imports/utils/Utils.qml]: updated

**Comment by jrainville** [comment]
@abhi-status the problems you found should be fixed.

This PR now depends on a StatusQ PR: https://github.com/status-im/StatusQ/pull/561

File: src/app/modules/main/app_search/module.nim
Comment (jrainville) [inline src/app/modules/main/app_search/module.nim:114]: I had to add this condition because calling `setItems` repeatetly while inside a community caused a crash. `beginResetModel` just failed for no reason sometimes, but reproducibly.

I'm still unsure why, so this condition above fixes it, because it seems like it was the repeated reset of the model that caused issues.

@osmaczko since you're the one that created the code here, can look at it and let me know if you think it might cause issues? I tested it it out and it seemed to work well, even if I add new channels, the search seemed to still work.

Comment (saledjenic) [reply src/app/modules/main/app_search/module.nim:114]: @jrainville this check `if (items.len != self.view.locationMenuModel().len):` as I see will introduce new issue, cause every time one runs "app search popup" we're building a location menu based on the current state. With this check if some community/chat change image/color or new the length will be the same, but the content won't be accurate.

A crash you mentioned is most likely caused on the qml side, cause that popup menu is being built from the model and some submenus/actions are added dynamically, maybe we should search in that direction for the crash reason. Maybe we should also experiment a bit with signals sent when we're setting a new set of items.

File: ui/app/AppMain.qml
Comment (saledjenic) [inline ui/app/AppMain.qml:778]: @jrainville typo `so this does ot work to close`.

**Review by saledjenic — CHANGES_REQUESTED** [review]

File: src/app/modules/main/app_search/module.nim
Comment (osmaczko) [reply src/app/modules/main/app_search/module.nim:114]: Hm..:question: I am not the author of this code :thinking:

Comment (jrainville) [reply src/app/modules/main/app_search/module.nim]: @saledjenic alright I finally figured out the issue. See AppSearch.qml. The problem was that the old list was not emptied correctly , leaving some remains, so if you had multiple communities, there ended up being a wrong memory access. Also, the duplicated community, when hovered crashed because again, there was no model for it.

@osmaczko my bad, I had forgotten that it was Sale who created it.

**Review by saledjenic — APPROVED** [review]

**Comment by abhi-status** [comment]
hey @jrainville 
Could you please check the latest PR builds here ?
I cant seem to launch the app install on both Windows and Ubuntu.

File: ui/app/AppSearch.qml
Comment (osmaczko) [inline ui/app/AppSearch.qml:33]:
```suggestion
            while (searchPopupMenu.takeItem(searchPopupMenu.NUM_DEFAULT_ITEMS)) {
                // Delete the item right after the default items
                // If takeItem returns null, it means there was nothing to remove
            }
```

Comment (osmaczko) [reply ui/app/AppSearch.qml:33]: Could be moved to `StatusSearchLocationMenu.qml` ?
`onModelAboutToBeReset: searchPopupMenu.removeCustomItems()`

Btw. it feels wrong to remove items manually :thinking:  but it's for another discussion

Comment (jrainville) [reply ui/app/AppSearch.qml:33]: > Could it be moved to StatusSearchLocationMenu.qml ?

This was @saledjenic 's code. What do you think @saledjenic, can the reset code be moved to `StatusSearchLocationMenu.qml`?

**Comment by jrainville** [comment]
@abhi-status hopefully now Jenkins will build

File: ui/app/AppSearch.qml
Comment (saledjenic) [reply ui/app/AppSearch.qml]: @jrainville in real that's @alexandraB99 code :) (that qml part). 

I am thinking now... and I am not sure why would we move that connection to `StatusSearchLocationMenu` component since that is actually a use case defined in our app, how we want to use `StatusSearchLocationMenu` component which is part of the `StatusQ`. Moving it there we will somehow limit it, or we can add another flag for for example to enable/disable that connection, but that just inflates a component.

All in all, I wouldn't move it, but also if you two decide to move it I am ok with that as well.

**Comment by abhi-status** [comment]
This is working now. 
However, due to issues addressed by #4866 , I am unsure if the user is taken to the correct channel / chat view when the user selects an item in the Activity panel. I think this can be tested reliably only once both #4866 and this PR are merged to Master.
Approving now.

**Review by abhi-status — APPROVED** [review]

File: ui/app/AppSearch.qml
Comment (osmaczko) [reply ui/app/AppSearch.qml]: I didn't mean to move the connection but only the part related to deleting custom items. Since `NUM_DEFAULT_ITEMS` is already in `StatusSearchLocationMenu` component it would make sense to also have `removeCustomItems()` function there. 
```
function removeCustomItems {
    while (searchPopupMenu.takeItem(searchPopupMenu.NUM_DEFAULT_ITEMS)) {}
}
```

Comment (saledjenic) [reply ui/app/AppSearch.qml]: @osmaczko yes, that sounds reasonable.

**Pull request merged**
